diff -urN epsilon-0.3.0.007.org/src/include/epsilon_private.h epsilon-0.3.0.007/src/include/epsilon_private.h
--- epsilon-0.3.0.007.org/src/include/epsilon_private.h	2006-03-28 11:10:14.000000000 +0200
+++ epsilon-0.3.0.007/src/include/epsilon_private.h	2007-01-04 18:30:26.093152250 +0100
@@ -41,7 +41,7 @@
 Epsilon_Message *epsilon_message_new(int clientid, char *path, char *dst, int status);
 
 int epsilon_ipc_client_send(Ecore_Con_Client *cl, Epsilon_Message *msg);
-int epsilon_ipc_server_send(Ecore_Con_Client *cl, Epsilon_Message *msg);
+int epsilon_ipc_server_send(Ecore_Con_Server *cl, Epsilon_Message *msg);
 
 Epsilon_Message *epsilon_ipc_consume(Epsilon_Ipc_End *end);
 
